web/ui/react-app/src/utils/index.ts

```diff
@@ -243,9 +243,21 @@ export const encodePanelOptionsToQueryString = (panels: PanelMeta[]): string =>
   return `?${panels.map(toQueryString).join('&')}`;
 };
 
+export const setQuerySearchFilter = (expr: string) => {
+  window.history.pushState({}, '', `?expr=${expr}`);
+};
+
+export const getQuerySearchFilter = (): string => {
+  const locationSearch = window.location.search;
+  const params = new URLSearchParams(locationSearch);
+  const search = params.get('expr') || '';
```

Comment:
I noticed this is still `expr` and not `search`, as I suggested earlier:

```suggestion
  const search = params.get('search') || '';
```

(needs a change in `setQuerySearchFilter` as well then)

Reply:
ready! :) i'm sorry thanks